### ISSUE TITLE
[REVIEW]: Neutron Imaging Normalization Library

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#815]
**Submitting author:** @jeanbilheux (<a href="http://orcid.org/0000-0003-2172-6487">Jean Bilheux</a>)
**Repository:** <a href="https://github.com/scikit-beam/NeuNorm" target ="_blank">https://github.com/scikit-beam/NeuNorm</a>
**Version:** 1.4.0
**Editor:** @arfon
**Reviewer:** @stuartcampbell
**Archive:** Pending

## Status

[![status](http://joss.theoj.org/papers/11e167ffb967ebde52213327da6d7323/status.svg)](http://joss.theoj.org/papers/11e167ffb967ebde52213327da6d7323)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/11e167ffb967ebde52213327da6d7323"><img src="http://joss.theoj.org/papers/11e167ffb967ebde52213327da6d7323/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/11e167ffb967ebde52213327da6d7323/status.svg)](http://joss.theoj.org/papers/11e167ffb967ebde52213327da6d7323)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/scikit-beam/NeuNorm" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@stuartcampbell, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @arfon know.

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/scikit-beam/NeuNorm">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (1.4.0)?
- [ ] **Authorship:** Has the submitting author (@jeanbilheux) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks. @stuartcampbell it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00815/joss.00815/10.21105.joss.00815.pdf)

**Comment by stuartcampbell** [comment]
@arfon Sorry for the delay in getting back to this.

I was just reading the conflict of interest, and I notice that it has a definition of recent as 4 years… I used to be in the same group as Jean 2 years ago, but I was never involved in any of the work he was doing in the area related to this publication.  This will be the first time I will have actually seen or looked at any of this code/work. 

Also, I am an owner of the ‘scikit-beam’ repo that this software is being hosted under, but this is just an umbrella org to try and co-locate any python based software from synchrotron and neutron sources.

I just wanted to mention these in case it rules me out of being a reviewer.

**Comment by stuartcampbell** [comment]
@arfon Just a quick question relating to the version listed here for review (which is 1.4.0), the latest version is 1.4.1.  @jeanbilheux has created all the releases for 1.4.1 but not 1.4.0.  Should I worry about this and ask for the older releases to be tagged ?

**Comment by arfon** [comment]
If @JeanBilheux is OK with it, we should just update the review to make it clear that `v1.4.1` is the version being reviewed here.

**Comment by JeanBilheux** [comment]
Sorry about that guys, but yes, 1.4.1 is the version to test. Thanks

**Comment by stuartcampbell** [comment]
@JeanBilheux - I've think I'm done with the review.  All in all looks good, I would just say to just add some simple contributing guidelines and update the references in the paper to include the software you are making use of (basically what you already have in you .bib file) - Then I think we are good to go!  Cheers
Stu

**Comment by JeanBilheux** [comment]
I updated the references in the README.md as well as the paper.md files. I added a CONTRIBUTING.md file as well.

**Comment by JeanBilheux** [comment]
With all those changes, should I make a new release version 1.4.2 ?

**Comment by arfon** [comment]
> With all those changes, should I make a new release version 1.4.2 ?

Yes, that sounds like a good idea.

**Comment by stuartcampbell** [comment]
@whedon command

**Comment by stuartcampbell** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00815/joss.00815/10.21105.joss.00815.pdf)

**Comment by stuartcampbell** [comment]
@arfon I'm happy with the changes made and have finished my review

**Comment by arfon** [comment]
Thanks @stuartcampbell! @JeanBilheux - At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by arfon** [comment]
Friendly reminder we're waiting on you here @JeanBilheux ☝️

**Comment by JeanBilheux** [comment]
Good morning. I’m glad you reminded me as I was waiting too   ☺

I will work on it

Thanks

Jean

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.1414282 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.1414282" target="_blank">10.5281/zenodo.1414282</a> is the archive.
